tests/test_flow/test_persistence.py

```diff
@@ -1208,3 +1208,53 @@ def x_y():
     assert builder.build().get("x") == 1
     assert builder.build().get("y") == 2
     assert counter.times_called() == 1
+
+
+@pytest.mark.needs_parallel
+def test_entries_needed_in_memory(builder, make_counter):
+    """
+      ----> B ---->
+    E               A
+      ----> C ---->
+
+    A depends on B and C
+    B and C depends on D
+    Only C is persisted
+
+    This structure is complex because it also tests for the case where
+    a deferred entity is moved back to pending stage. In this case,
+    D is deferred by C but then is required in memory by B.
+    """
+    builder.set("core__persist_by_default", False)
```

Comment:
Personally I think I would find this clearer if we just used `@persist(False)` for the entities we don't want to persist.